### PR TITLE
Improve test coverage of convert module

### PR DESCRIPTION
**Pull request by rossbar**
Try to improve test coverage of the `networkx/convert.py`. 

Todo:
 - Remove warnings on ImportError for numpy/scipy/pandas etc.
   * Tentatively in favor of leaving this as-is.
 - [x] Resolve expected behavior of `to_dict_of_dicts` for multigraphs (see #4305)

### CODE REVIEW

File: networkx/convert.py [hunk 1]
```diff
@@ -78,14 +78,14 @@ def to_networkx_graph(data, create_using=None, multigraph_input=False):
                 create_using=create_using,
                 multigraph_input=data.is_multigraph(),
             )
-            if hasattr(data, "graph"):  # data.graph should be dict-like
-                result.graph.update(data.graph)
-            if hasattr(data, "nodes"):  # data.nodes should be dict-like
```
Comment (rossbar) [inline networkx/convert.py:83]: Is it possible to have a Graph object that doesn't have `graph` and `nodes` attributes? The [documentation on subclassing](https://networkx.org/documentation/latest/reference/classes/graph.html#Subclasses_(Advanced)) makes it seem like subclasses must always have these attributes. If that's the case then I think these `if` s can be removed. If not, then we can use the counter example in the tests!

Comment (dschult) [reply networkx/convert.py:83]: I think these checks allow someone to create a class that only holds the adj attribute info and doesn't include node attributes or graph attributes. I don't think that is likely to happen -- probably never has happened.
This change looks good to me!

File: networkx/tests/test_convert.py [hunk 2]
```diff
@@ -279,3 +279,23 @@ class Custom(nx.Graph):
...
+def test_to_networkx_graph_non_edgelist():
+    invalid_edgelist = [1, 2, 3]
+    with pytest.raises(nx.NetworkXError, match="Input is not a valid edge list"):
+        nx.to_networkx_graph(invalid_edgelist)
+
+
+def test_to_dict_of_dicts_with_edge_data():
+    G = nx.Graph([(0, 1)])
+    expected = {0: {1: {"weight": 1.0}}, 1: {0: {"weight": 1.0}}}
+    assert nx.to_dict_of_dicts(G, edge_data={"weight": 1}) == expected
```
Comment (dschult) [inline networkx/tests/test_convert.py:293]: edge_data should not be a container! The edges will end up all sharing the same container.

I think a test using the value `1` is a better choice. Also, see [this comment](https://github.com/networkx/networkx/issues/4305#issuecomment-719970727)

Comment (rossbar) [reply networkx/tests/test_convert.py:293]: Thanks for the feedback in the linked comment. I've moved the tests related to `edge_data` to #4321 which deals specifically with `edge_data`.

**Comment by rossbar** [comment]
This ups the coverage of convert to 99%. The remaining "uncovered" branches raise ImportWarning when default dependencies are not found. My feeling is to leave these in for now, and address `try/except ImportErrors` of the default dependencies in a separate PR (potentially post 2.6 after we see if there's any feedback from adding numpy/scipy et. al. as dependencies)

However, if you'd like to just remove these three try/excepts here to get the coverage to 100% I'm fine with that!

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**